src/main/java/use_cases/play_card_use_case/PlayCardInteractor.java

```diff
@@ -0,0 +1,198 @@
+        this.gameState = gameState;
+        this.playCardPresenter = playCardPresenter;
+    }
+
+
+    /**
+     * Fetch the card object from the player, and play if it can be played
+     * without target selection (a second input from the player), or tell the
+     * presenter the possible targets and let presenter open a target selection
+     * screen.
+     *
+     * @param cardId the id of the card, given from the controller's layer.
+     */
+    @Override
+    public PlayCardOutputModel playCard(int cardId) {
+        Player player = gameState.getCurrentPlayer();
+        Player opponent = gameState.getOpposingPlayer();
+        Card card = player.getCardById(cardId);
+
+        if (!(card instanceof Playable)) {
+            return playCardPresenter.displayErrorMessage("This card cannot be played directly.");
+        }
+        if (card instanceof StructureCard) {
+            player.playCard(cardId);
+            player.setStructure((StructureCard) card);
+            PlayCardOutputModel outputModel = getOutputModel();
+            return playCardPresenter.updateGameScreen(outputModel);
+        }
+        // The only other case is the card being Action, so we can safely cast it.
+        ActionCard action = (ActionCard) card;
+        List<Integer> targetIds = new ArrayList<>();
+        switch (action.getTargetType()) {
+            case SINGLE_ENEMY:
+                targetIds.addAll(getCreatureIds(opponent.getCreatures()));
+                break;
+            case SINGLE_FRIENDLY:
+                targetIds.addAll(getCreatureIds(player.getCreatures()));
+                break;
+            case ANY:
+                targetIds.addAll(getCreatureIds(player.getCreatures()));
+                targetIds.addAll(getCreatureIds(opponent.getCreatures()));
+                break;
+        }
+        if (targetIds.size() != 0) {
+            TargetModel targetModel = new TargetModel(cardId, targetIds);
+            return playCardPresenter.showTargetSelectionScreen(targetModel);
+        }
+
+        // If we didn't return in the above, we must be arriving at an action
+        // that can be immediately spent. So we *spend* it from the player's hand
+        // first, so that any cards drawn would not mistakenly get discarded
+        player.playCard(cardId);
+        switch (action.getTargetType()) {
+            case FRIENDLY:
+                useAction(action, player.getCreatures());
+                break;
+            case ENEMY:
+                useAction(action, opponent.getCreatures());
+                break;
+            case ALL:
+                useAction(action, player.getCreatures());
+                useAction(action, opponent.getCreatures());
+                break;
+            case SELF:
+                useAction(action, player);
+                break;
+            case OPPONENT:
+                useAction(action, opponent);
+                break;
```

Comment:
I think these two cases can be handled by a new method. It might be helpful to understand the <playCard> method.

Reply:
yup, we'll need to try optimize this in a new branch for refactoring code